### PR TITLE
[feature-#14449] add secret in k8s Task

### PR DESCRIPTION
**Pull request by fuchanghai**
## Purpose of the pull request
- this close this issue : #14449 
## Verify this pull request

This pull request is code cleanup without any test coverage.

### CODE REVIEW

**Comment by fuchanghai** [comment]
@zhuangchong PTAL

**Review by songjianet — DISMISSED** [review]

File: dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/k8s/K8sTaskMainParameters.java [hunk 1]
```diff
@@ -32,6 +32,7 @@ public class K8sTaskMainParameters {
     private String image;
     private String command;
     private String args;
+    private String secret;
```
Comment (qingwli) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/k8s/K8sTaskMainParameters.java]: Please rename this field

**Review by qingwli — CHANGES_REQUESTED** [review]

File: dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/k8s/K8sTaskMainParameters.java (hunk 1 above)
Comment (fuchanghai) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/k8s/K8sTaskMainParameters.java]: done

**Comment by qingwli** [comment]
Have you tested if the pull secret is an empty value case?

**Comment by qingwli** [comment]
Please fix the ut

**Comment by fuchanghai** [comment]
hi @qingwli I will solve these problems over the weekend

**Comment by fuchanghai** [comment]
> Please fix the ut

done

**Comment by fuchanghai** [comment]
> Have you tested if the pull secret is an empty value case?

hi @qingwli  now, if ```pullSecret ``` value equals ```null ``` will not be putted into ```job``` cc @Gallardot

File: dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/k8s/impl/K8sTaskExecutor.java [hunk 2]
```diff
@@ -184,8 +187,11 @@ public Job buildK8sJob(K8sTaskMainParameters k8STaskMainParameters) {
                 .endTemplate()
                 .withBackoffLimit(retryNum)
                 .endSpec();
-
-        return jobBuilder.build();
+        job = jobBuilder.build();
+        if (!StringUtils.isEmpty(pullSecret)) {
+            job.getSpec().getTemplate().getSpec()
+                    .setImagePullSecrets(singletonList(new LocalObjectReference(pullSecret)));
+        }
```
Comment (Gallardot) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/k8s/impl/K8sTaskExecutor.java]: I'm a bit confused about this code. Why not use `withImagePullSecrets` before the job build?

**Review by Gallardot — COMMENTED** [review]
Rest LGTM.

File: dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/k8s/impl/K8sTaskExecutor.java (hunk 2 above)
Comment (qingwli) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/k8s/impl/K8sTaskExecutor.java]: Looks like we can use this usage avoid empty case
  ```
   .withCommand(commands.size() == 0 ? null : commands)
                .withArgs(args.size() == 0 ? null : args)
  ```

Comment (fuchanghai) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/k8s/impl/K8sTaskExecutor.java]: done

**Comment by qingwli** [comment]
Run 'mvn spotless:apply' to fix these violations.

**Comment by fuchanghai** [comment]
> mvn spotless:apply

done

**Review by qingwli — APPROVED** [review]
LGTM

**Pull request merged**